Build error with Boost 1.75.0

`telnetpp` doesn't compile with MSVC compiler and Boost 1.75.0.
Error:
```
[6/24] "C:\PROGRA~2\Microsoft Visual Studio\2019\Enterprise\VC\Tools\MSVC\14.27.29110\bin\Hostx64\x64\cl.exe"   /TP -Dtelnetpp_EXPORTS -ID:\buildtrees\telnetpp\src\c050d4c659-52a444ec23.clean\include -ID:\installed\x64-windows\include /nologo /DWIN32 /D_WINDOWS /W3 /utf-8 /GR /EHsc /MP  /D_DEBUG /MDd /Z7 /Ob0 /Od /RTC1  -std:c++14 /showIncludes /FoCMakeFiles\telnetpp.dir\src\options\msdp\variable.cpp.obj /FdCMakeFiles\telnetpp.dir\ /FS -c D:\buildtrees\telnetpp\src\c050d4c659-52a444ec23.clean\src\options\msdp\variable.cpp
FAILED: CMakeFiles/telnetpp.dir/src/options/msdp/variable.cpp.obj 
"C:\PROGRA~2\Microsoft Visual Studio\2019\Enterprise\VC\Tools\MSVC\14.27.29110\bin\Hostx64\x64\cl.exe"   /TP -Dtelnetpp_EXPORTS -ID:\buildtrees\telnetpp\src\c050d4c659-52a444ec23.clean\include -ID:\installed\x64-windows\include /nologo /DWIN32 /D_WINDOWS /W3 /utf-8 /GR /EHsc /MP  /D_DEBUG /MDd /Z7 /Ob0 /Od /RTC1  -std:c++14 /showIncludes /FoCMakeFiles\telnetpp.dir\src\options\msdp\variable.cpp.obj /FdCMakeFiles\telnetpp.dir\ /FS -c D:\buildtrees\telnetpp\src\c050d4c659-52a444ec23.clean\src\options\msdp\variable.cpp
D:\buildtrees\telnetpp\src\c050d4c659-52a444ec23.clean\include\telnetpp/options/msdp/variable.hpp(70): warning C4251: 'telnetpp::options::msdp::variable::name': class 'std::basic_string<telnetpp::byte,std::char_traits<telnetpp::byte>,std::allocator<telnetpp::byte>>' needs to have dll-interface to be used by clients of struct 'telnetpp::options::msdp::variable'
D:\buildtrees\telnetpp\src\c050d4c659-52a444ec23.clean\include\telnetpp/core.hpp(43): note: see declaration of 'std::basic_string<telnetpp::byte,std::char_traits<telnetpp::byte>,std::allocator<telnetpp::byte>>'
D:\buildtrees\telnetpp\src\c050d4c659-52a444ec23.clean\include\telnetpp/options/msdp/variable.hpp(71): warning C4251: 'telnetpp::options::msdp::variable::value': class 'boost::variant<telnetpp::options::msdp::string_value,telnetpp::options::msdp::array_value,boost::recursive_wrapper<telnetpp::options::msdp::table_value>>' needs to have dll-interface to be used by clients of struct 'telnetpp::options::msdp::variable'
D:\buildtrees\telnetpp\src\c050d4c659-52a444ec23.clean\include\telnetpp/options/msdp/variable.hpp(22): note: see declaration of 'boost::variant<telnetpp::options::msdp::string_value,telnetpp::options::msdp::array_value,boost::recursive_wrapper<telnetpp::options::msdp::table_value>>'
D:\installed\x64-windows\include\boost/move/detail/type_traits.hpp(968): error C2079: 'boost::move_detail::alignment_of_hack<T>::t1' uses undefined struct 'telnetpp::options::msdp::variable' 
...
```

More information you can find in PR: https://github.com/microsoft/vcpkg/pull/15090
Full build log: https://dev.azure.com/vcpkg/c1ee48cb-0df2-4ab3-8384-b1df5a79fe53/_apis/build/builds/46451/artifacts?artifactName=failure%20logs%20for%20x64-windows&api-version=6.0&%24format=zip

Ok, it seems that shared libraries do generate a lot of noise, but that isn't what actually broke in the full build linked.  That was here:

`D:\installed\x64-windows\include\boost/move/detail/type_traits.hpp(968): error C2079: 'boost::move_detail::alignment_of_hack<T>::t1' uses undefined struct 'telnetpp::options::msdp::variable'
        with
        [
            T=telnetpp::options::msdp::variable
        ]
D:\installed\x64-windows\include\boost/move/detail/type_traits.hpp(985): note: see reference to class template instantiation 'boost::move_detail::alignment_of_hack<T>' being compiled
        with
        [
            T=telnetpp::options::msdp::variable
        ]
D:\installed\x64-windows\include\boost/move/detail/type_traits.hpp(997): note: see reference to class template instantiation 'boost::move_detail::alignment_of_impl<T>' being compiled
        with
        [
            T=telnetpp::options::msdp::variable
        ]
D:\installed\x64-windows\include\boost/container/small_vector.hpp(417): note: see reference to class template instantiation 'boost::move_detail::alignment_of<T>' being compiled
        with
        [
            T=telnetpp::options::msdp::variable
        ]
D:\installed\x64-windows\include\boost/container/small_vector.hpp(549): note: see reference to class template instantiation 'boost::container::small_vector_base<T,Allocator,Options>' being compiled
        with
        [
            T=telnetpp::options::msdp::variable,
            Allocator=void,
            Options=void
        ]
D:\installed\x64-windows\include\boost/type_traits/is_nothrow_move_assignable.hpp(29): note: see reference to class template instantiation `

Oh I see it's still there: 

`D:\libs\yurybura\vcpkg\installed\x64-windows\include\boost/move/detail/type_traits.hpp(968,6): error C2079: 'boost::move_detail::alignment_of_hack<T>::t1' uses undefined struct 'telnetpp::options::msdp::variable' `

--- a/include/telnetpp/options/msdp/variable.hpp
+++ b/include/telnetpp/options/msdp/variable.hpp
@@ -4,7 +4,7 @@
 #include <boost/container/small_vector.hpp>
 #include <boost/variant.hpp>
 #include <iosfwd>
-#include <string>
+#include <vector>
 
 namespace telnetpp { namespace options { namespace msdp {
 
@@ -12,7 +12,7 @@ struct variable;
 
 using string_value = telnetpp::byte_storage;
 using array_value  = boost::container::small_vector<string_value, 4>;
-using table_value  = boost::container::small_vector<variable, 4>;
+using table_value  = std::vector<variable>;
 
 //* =========================================================================
 /// \class telnetpp::options::msdp::value_type

